src/rdkafka_broker.c

```diff
@@ -4156,26 +4156,28 @@ static int rd_kafka_broker_thread_main (void *arg) {
 	rd_kafka_broker_lock(rkb);
 	rd_kafka_broker_unlock(rkb);
 
-	rd_rkb_dbg(rkb, BROKER, "BRKMAIN", "Enter main broker thread");
+        rd_rkb_dbg(rkb, BROKER, "BRKMAIN", "Enter main broker thread");
 
-	while (!rd_kafka_broker_terminating(rkb)) {
+        int initial_token_unavailable = 0;
+
+        while (!rd_kafka_broker_terminating(rkb) && !initial_token_unavailable) {
                 int backoff;
 
         redo:
                 switch (rkb->rkb_state)
                 {
                 case RD_KAFKA_BROKER_STATE_INIT:
                         /* Check if there is demand for a connection
-                         * to this broker, if so jump to TRY_CONNECT state. */
```

Comment:
Avoid whitespace diffs (and these actually break intended indention),
there's a bunch of them below as well.

Reply:
👍 All pure-whitespace diffs are now reverted, everywhere.